tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +91,494 @@ mvn package
 
 ```java
-Profile profile = userProfile.getProfile();
-profileCol.insert(profile.getPid(), profile);
-return ResponseEntity.status(HttpStatus.CREATED).body(profile);
-```
+public interface AirlineRepository {
+    Airline findById(String id);
 
-> *from controllers/ProfileController.java*
+    Airline save(Airline airline);
 
-## GET a Profile by Key
+    Airline update(String id, Airline airline);
 
-Navigate to the getProfile method in the ProfileController file in the controllers folder.  We only need the profile ID `pid` or our `Key` from the user to retrieve a particular profile document using a basic key-value operation which is passed in the method signature as a string.  Since we created the document with a unique key we can use that key to find the document in the scope and collection it is stored in.
+    void delete(String id);
+
+    List<Airline> findAll(int limit, int offset);
+
+    List<Airline> findByCountry(String country, int limit, int offset);
+
+    List<Airline> findByDestinationAirport(String destinationAirport, int limit, int offset);
+}
+
+@Repository
+public class AirlineRepositoryImpl implements AirlineRepository {
+
+    private final Cluster cluster;
+    private final Collection airlineCol;
+    private final CouchbaseConfig couchbaseConfig;
+
+    public AirlineRepositoryImpl(Cluster cluster, Bucket bucket, CouchbaseConfig couchbaseConfig) {
+        this.cluster = cluster;
+        this.airlineCol = bucket.scope("inventory").collection("airline");
+        this.couchbaseConfig = couchbaseConfig;
+    }
+
+    @Override
+    public Airline findById(String id) {
+        return airlineCol.get(id).contentAs(Airline.class);
+    }
+
+    @Override
+    public Airline save(Airline airline) {
+        airlineCol.insert(airline.getId(), airline);
+        return airline;
+    }
+
+    @Override
+    public Airline update(String id, Airline airline) {
+        airlineCol.replace(id, airline);
+        return airline;
+    }
+
+    @Override
+    public void delete(String id) {
+        airlineCol.remove(id);
+    }
+
+}
 
-```java
-Profile profile = profileCol.get(pid).contentAs(Profile.class);
-return ResponseEntity.status(HttpStatus.OK).body(profile);
 ```
 
-> *from getProfile method of controllers/ProfileController.java*
+## Mapping Workflow
+
+### GET Mapping Workflow
+
+`@GetMapping("/{id}")`
+
+The GET mapping is triggered when a client sends an HTTP GET request to `/api/v1/airline/{id}` where `{id}` is the unique identifier of the airline.
+
+1. The `AirlineController` receives the request and extracts the `id` from the URL path.
+2. It then calls the `getAirlineById` method of the `AirlineService`, passing the extracted `id` as a parameter.This function internally calls [airlineCol.get()](https://docs.couchbase.com/java-sdk/current/howtos/kv-operations.html#retrieving-documents) to retrieve the airline from the database.
```

Comment:
I would change airlineCol.get() to get() in collection defined by the SDK.

Reply:
changed it to get()